library/bignum_mod.c

```diff
@@ -141,8 +141,15 @@ int mbedtls_mpi_mod_modulus_setup( mbedtls_mpi_mod_modulus *m,
                                    mbedtls_mpi_mod_ext_rep ext_rep,
                                    mbedtls_mpi_mod_rep_selector int_rep )
 {
-    int ret = 0;
+    int ret = MBEDTLS_ERR_MPI_BAD_INPUT_DATA;
+    size_t zero_c;
 
+    for( zero_c = 0; zero_c < p_limbs; zero_c++ )
```

Comment:
Readability, debuggability and maintainability suggestion: put the code in an auxiliary function and write
```
    if( mpi_core_is_zero( p, p_limbs ) )
        return( MBEDTLS_ERR_MPI_BAD_INPUT_DATA );
```

That way:

* It's immediately clear to the reader what is being checked: you can tell from the function name, you don't need to work it out by reading the loop.
* If see that a function returns `MBEDTLS_ERR_MPI_BAD_INPUT_DATA` and we suspect this is the cause, there's a place we can put a breakpoint to check it.
* If we ever need the same function again, it's readily available, so we run less risk of duplicating the code.